File: 0238 product-of-array-except-self/solution.go
func productExceptSelf(nums []int) []int {
    n:=len(nums)
    var ans = make([]int,n)
    ans[0]=1
    for i:=1;i<n;i++ {
        ans[i]=ans[i-1]*nums[i-1]
    }
    right:=1
    for i:=n-1;i>=0;i-- {
        ans[i]*=right
        right*=nums[i]
    }
    return ans
}

File: 0238 product-of-array-except-self/solution.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
       vector<int> ans(nums.size(),1);
        ans[0] = 1;
        for(int i=1;i<nums.size();i++) {
            ans[i] = ans[i-1] * nums[i-1];
        }
        int suff = 1;
        for(int n = ans.size() - 1 ; n>=0; n--) {
            ans[n] *= suff;
            suff*=nums[n];
        }
        return ans;
    }
};
